tests/test_operation.py

```diff
@@ -19,57 +20,61 @@ def test_operation(
     strategy.tend()
 
     # withdrawal
-    vault.withdraw({"from": accounts[0]})
-    assert pytest.approx(token.balanceOf(accounts[0]), rel=RELATIVE_APPROX) == amount
+    vault.withdraw({"from": user})
+    assert pytest.approx(token.balanceOf(user), rel=RELATIVE_APPROX) == user_balance_before
 
 
 def test_emergency_exit(
-    accounts, token, vault, strategy, strategist, amount, RELATIVE_APPROX
+    accounts, token, vault, strategy, user, strategist, amount, RELATIVE_APPROX
 ):
     # Deposit to the vault
-    token.approve(vault.address, amount, {"from": accounts[0]})
-    vault.deposit(amount, {"from": accounts[0]})
-    strategy.harvest()
+    token.approve(vault.address, amount, {"from": user})
+    vault.deposit(amount, {"from": user})
+    strategy.harvest({"from": strategist})
     assert pytest.approx(strategy.estimatedTotalAssets(), rel=RELATIVE_APPROX) == amount
 
     # set emergency and exit
     strategy.setEmergencyExit()
-    strategy.harvest()
+    strategy.harvest({"from": strategist})
     assert strategy.estimatedTotalAssets() < amount
 
 
 def test_profitable_harvest(
-    accounts, token, vault, strategy, strategist, amount, RELATIVE_APPROX
+    accounts, token, vault, strategy, user, strategist, amount, RELATIVE_APPROX, chain
 ):
     # Deposit to the vault
-    token.approve(vault.address, amount, {"from": accounts[0]})
-    vault.deposit(amount, {"from": accounts[0]})
+    token.approve(vault.address, amount, {"from": user})
+    vault.deposit(amount, {"from": user})
     assert token.balanceOf(vault.address) == amount
 
-    # harvest
-    strategy.harvest()
-    assert pytest.approx(strategy.estimatedTotalAssets(), rel=RELATIVE_APPROX) == amount
+    # Harvest 1: Send funds through the strategy
+    strategy.harvest({"from": strategist})
+    assert strategy.estimatedTotalAssets() == amount
```

Comment:
You should keep the approximation, the base strategy is fairly simple but harvest might deposit into a contract that does some complex calculation and rounding, the test will fail.

Reply:
OK